src/search/include/search/create_controller.h

```diff
@@ -72,15 +71,14 @@ get_constraints_from_time_successor(
  */
 template <typename LocationT, typename ActionT>
 std::multimap<ActionT, std::multimap<std::string, automata::ClockConstraint>>
-get_constraints_from_outgoing_actions(
+get_constraints_from_outgoing_action(
   const std::set<search::CanonicalABWord<LocationT, ActionT>> canonical_words,
-  const std::set<std::pair<search::RegionIndex, ActionT>> &   actions,
+  const std::pair<search::RegionIndex, ActionT> &             timed_action,
   search::RegionIndex                                         K)
 {
 	std::map<ActionT, std::set<search::RegionIndex>> good_actions;
-	for (const auto &[region_increment, action] : actions) {
-		good_actions[action].insert(region_increment);
-	}
+	// TODO merging of the constraints is broken because we now get only a single action.
```

Comment:
Not sure whether this works, but would it be an option to store an interval/set of region increments instead? What is probably provable: It cannot be, that an action is enabled for several disjoint time intervals in the same location without leaving the location inbetween (remains to be shown, but I am pretty sure this holds). If that holds, we can later store the convex closure of the increments, i.e., a time interval.

Reply:
Not sure if it holds, you can have two transitions with different guards, e.g.,
```
s0 -> a / x < 1 / {} -> s1
s0 -> a / x > 2 / {} -> s1
```

Wouldn't that be a counter example to the claim?

Reply:
Good point - this holds only for individual transitions, i.e., a transition can at most switch once from not being enable to being enabled and once from being enabled to not being enabled. This means that we can only merge per transition - those are not available at that point, right?

Reply:
So I think finding adjacent transitions is not the problem, but we may have a problem if they lead to different successors (due to the regions being different), then we'd need to compare sub-trees.

Reply:
I'd suggest we move the merging of constraints to a follow-up PR? It may also no longer be necessary once we only take the first controller action and ignore the rest (I will follow up with this)

Reply:
Yes, you are right. Let's keep things simple, for a change :)

Reply:
Sounds good! Can you re-approve if there are no further open issues? :)